pym/gentoolkit/eclean/search.py

```diff
@@ -332,6 +332,31 @@ def _non_destructive(
             deprecated.update(_deprecated)
         return pkgs, deprecated
 
+    def git_check(self, distdir):
+        """Checks $DISTDIR/git3-src for checkouts which are not in the vardb"""
+        git_src = os.path.join(distdir, "git3-src")
+        expected_dirs = set()
+        for i in set(self.vardb.cpv_all()):
+            if "live" in self.vardb.aux_get(i, ["PROPERTIES"]):
+                try:
+                    # try to get the dir names of the cloned
+                    # repos from the environment file.
+                    vcs_dir = {
+                        i.split("=")[1].strip('"')
+                        for i in (
+                            self.vardb._aux_env_search(i, ["EVCS_STORE_DIRS"])[
+                                "EVCS_STORE_DIRS"
+                            ]
+                            .strip("()")
+                            .split()
```

Comment:
This won't handle whitespace correctly. The whole point of using an array is to handle it correctly.

Reply:
The eclass currently does not use whitespace in `repo_name`, so this is problem in theory only.

Are you aware of any way to properly parse a bash array in Python?

Reply:
> The eclass currently does not use whitespace in `repo_name`, so this is problem in theory only.

The variable is supposed to contain the full path.

> Are you aware of any way to properly parse a bash array in Python?

Perhaps `shlex` would be good enough.

Reply:
> The variable is supposed to contain the full path.

No, it's not. The full path is useless since the user could change DISTDIR at any time.

Reply:
> Perhaps shlex would be good enough.

Yeah, `shlex.split()` will work after stripping the parens.